### ISSUE TITLE
Upgrade to zlib v1.2.12 in CPython binary releases

### ISSUE THREAD

**Issue by gpshead** [python/cpython#91350]
BPO | [47194](https://bugs.python.org/issue47194)
--- | :---
Nosy | @gpshead, @pfmoore, @tjguk, @ned-deily, @ambv, @zware, @zooba, @pablogsal, @miss-islington
PRs | <li>python/cpython#32241</li><li>python/cpython#32248</li><li>python/cpython#32249</li><li>python/cpython#32250</li><li>python/cpython#32251</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2022-04-01.19:25:42.678>
labels = ['3.8', '3.9', 'release-blocker', 'extension-modules', '3.7', 'type-security', '3.10', '3.11', 'OS-windows']
title = 'Upgrade to zlib v1.2.12 in CPython binary releases'
updated_at = <Date 2022-04-04.03:27:26.636>
user = 'https://github.com/gpshead'
```

bugs.python.org fields:
```python
activity = <Date 2022-04-04.03:27:26.636>
actor = 'ned.deily'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Extension Modules', 'Windows']
creation = <Date 2022-04-01.19:25:42.678>
creator = 'gregory.p.smith'
dependencies = []
files = []
hgrepos = []
issue_num = 47194
keywords = ['patch']
message_count = 5.0
messages = ['416510', '416552', '416555', '416556', '416651']
nosy_count = 9.0
nosy_names = ['gregory.p.smith', 'paul.moore', 'tim.golden', 'ned.deily', 'lukasz.langa', 'zach.ware', 'steve.dower', 'pablogsal', 'miss-islington']
pr_nums = ['32241', '32248', '32249', '32250', '32251']
priority = 'release blocker'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'security'
url = 'https://bugs.python.org/issue47194'
versions = ['Python 3.7', 'Python 3.8', 'Python 3.9', 'Python 3.10', 'Python 3.11']
```

</p></details>

**Comment by vstinner** [comment]
> It seems like zlib is still outdated in Python 3.8.

There is https://github.com/python/cpython/pull/32250 but the Docs CI failed. I removed/added back the "backport to 3.8" label on https://github.com/python/cpython/pull/32241 and it updated the 3.8 PR.

**Comment by gpshead** [comment]
yep just waiting on that 3.8 PR being merged by its RM. we don't have a good way to indicate which branches an issue remains a release-blocker for.

**Comment by zware** [comment]
With the merge of #32250, this is now resolved in all active branches.  Does anyone have a handle on whether the severity of the CVE is high enough to warrant releases?

**Comment by ambv** [comment]
Since we only ship our own zlib with installers and 3.7-3.8 aren't shipping installers anymore, I don't think we need a security release for *that*.  The only other change in 3.8 is a sqlite segfault fix so I think I'll pass on releasing a 3.8 at this time.

**Comment by ned-deily** [comment]
> Since we only ship our own zlib with installers and 3.7-3.8 aren't shipping installers anymore, I don't think we need a security release for _that_.

Likewise with 3.7 so time to close this. Thanks, everyone!
